src/Atmos/Model/boundaryconditions.jl

```diff
@@ -12,6 +12,7 @@ export AtmosBC,
     PrescribedEnergyFlux,
     BulkFormulaEnergy,
     Impermeable,
+    ImpermeablePrecipitation,
```

Comment:
I'm confused here by what this means. Precipitation should have an outflow condition, which means that the surface is permeable to it. (Maybe it's just a matter of terminology.)

Reply:
I admit I'm confused by boundary conditions. I think the implemented BC would result in having `rho q_rai = 0` in the ghost points and no additional conditions on the velocity field. So this would mean free outflow at the bottom and potentially inflow of zeros at the top and sides? @charleskawczynski - am I reading it right?

In general, I'm not sure if I should do anything more to make the BC consistent between what we do for air velocity and what happens for rain. Rain has its own velocity equal to the sum of air velocity and terminal velocity.

Reply:
Mathematically, what is the intended BC? When I read `Impermeable`, I think zero Neumann BCs. I think it'd be a good idea to loop in those who better understand `boundary_state!`. cc @jkozdon @blallen

Reply:
This is a BC on an advective flux. We want an outgoing condition on the falling precipitation. Usually outgoing conditions are hard to implement, but here it should be reasonably straightforward: The normal component of the general velocity field is zero at the surface; the normal component of the terminal velocity, which advects the rain relative to the air, should have no gradients at the surface (hence homogeneous Neumann conditions on this component).

Reply:
Based on my understanding, and using the heat tutorial as an example, we should be setting `diff⁺.∇q_rai = 0` in `atmos_precipitation_normal_boundary_flux_second_order!` which I think should result in `n̂⋅∇q_rai = 0`. That said, I don't see the same thing done in `atmos_moisture_normal_boundary_flux_second_order!`. @jkozdon can you weigh in?

Reply:
We should do this too, but it is not the boundary condition in question here. Setting the diffusive fluxes to zero at the boundary just sets the normal SGS flux to zero. The question here is about the advective fluxes (first-order fluxes).